LocalMongoDBService/LocalMongoDBService/LocalMongoDBService.swift

```diff
@@ -0,0 +1,168 @@
+import Foundation
+import MongoMobile
+import os
+import StitchCore
+import StitchCore_iOS
+
+/// Commands associated with reporting battery level
+/// to MongoMobile
+private enum BatteryLevelCommand: String {
+    case mongoCommand = "setBatteryLevel"
+    case batteryLevelLow = "low"
+    case batteryLevelNormal = "normal"
+}
+
+/// Current state of the UIDevice battery
+private enum BatteryState {
+    case low, normal
+}
+
+/// Commands associated with reporting memory level
+/// to MongoMobile
+private enum TrimMemoryCommand: String {
+    case mongoCommand = "trimMemory"
+    case aggressiveLevel = "aggressive"
+}
+
+/// Name of admin database to communicate commands to
+private let adminDatabaseName = "admin"
+
+/// Cached mongoClients associated to app ids
+private var localInstances = [String: MongoClient]()
+
+/// Local MongoDB Service Provider
+private final class MongoDBServiceClientProvider: ThrowingServiceClientProvider {
+    public typealias ClientType = MongoClient
+
+    /// Current battery level of this device between 0-100
+    private var batteryLevel: Float {
+        return UIDevice.current.batteryLevel
+    }
+
+    private var lastBatteryState: BatteryState
+
+    public init() {
+        // initialize mongo mobile
+        MongoMobile.initialize()
+
+        // enable battery monitoring
+        UIDevice.current.isBatteryMonitoringEnabled = true
+        if UIDevice.current.batteryLevel < 30 {
+            self.lastBatteryState = .low
+        } else {
+            self.lastBatteryState = .normal
+        }
+
+        // observe when the battery level changes
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(batteryLevelDidChange(_:)),
+            name: .UIDeviceBatteryLevelDidChange,
+            object: nil
+        )
+        // observe when memory warnings are received
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(didReceiveMemoryWarning(_:)),
+            name: .UIApplicationDidReceiveMemoryWarning,
+            object: nil
+        )
+    }
+
+    deinit {
+        // clean up mobile mongo
+        MongoMobile.close()
+    }
+
+    public func client(forService service: StitchService,
+                       withClientInfo clientInfo: StitchAppClientInfo) throws -> MongoClient {
+        objc_sync_enter(self)
+        defer { objc_sync_exit(self) }
+
+        // if client is cached, return it
+        if let client = localInstances[clientInfo.clientAppId] {
+            return client
+        }
+
+        // else, create a new client
+        let settings = MongoClientSettings(dbPath: clientInfo.dataDirectory.path)
```

Comment:
This still needs to be configured to have the app id path concatted on like java: 
```
final String dbPath = String.format("%s/local_mongodb/0/", dataDir);
    final MongoClient client =
        MongoClients.create(MongoClientSettings.builder().dbPath(dbPath).build());
```